### DISCUSSION TITLE
colmap gui does not work because of qt issues

### DISCUSSION THREAD

**Discussion by KansaiUser** [colmap/colmap#2386] [Q&A]
I am trying to use colmap for the first time with the docker file. I already downloaded the docker file and I do 

`docker run --gpus all -w /working -v /media/sensetime/cbe421fe-1303-4821-9392-a849bfdd00e2/Work/colmap/WorkingDirectory:/working -it colmap/colmap:latest`

with that I enter the container. I try then to do colmap gui and I get 

```
root@01fc4655de6b:/working# colmap gui
qt.qpa.xcb: could not connect to display 
qt.qpa.plugin: Could not load the Qt platform plugin "xcb" in "" even though it was found.
This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.

Available platform plugins are: eglfs, linuxfb, minimal, minimalegl, offscreen, vnc, xcb.

*** Aborted at 1706784366 (unix time) try "date -d @1706784366" if you are using GNU date ***
PC: @                0x0 (unknown)
*** SIGABRT (@0x1b) received by PID 27 (TID 0x7f379e015000) from PID 27; stack trace: ***
    @     0x7f37a3749046 (unknown)
    @     0x7f37a1dfc520 (unknown)
    @     0x7f37a1e509fc pthread_kill
    @     0x7f37a1dfc476 raise
    @     0x7f37a1de27f3 abort
    @     0x7f37a23a6ba3 QMessageLogger::fatal()
    @     0x7f37a29ad713 QGuiApplicationPrivate::createPlatformIntegration()
    @     0x7f37a29adc08 QGuiApplicationPrivate::createEventDispatcher()
    @     0x7f37a25d6b17 QCoreApplicationPrivate::init()
    @     0x7f37a29b0b70 QGuiApplicationPrivate::init()
    @     0x7f37a30c4ced QApplicationPrivate::init()
    @     0x560ef1d759ad colmap::RunGraphicalUserInterface()
    @     0x560ef1d381ab main
    @     0x7f37a1de3d90 (unknown)
    @     0x7f37a1de3e40 __libc_start_main
    @     0x560ef1d3e945 _start
Aborted (core dumped)
root@01fc4655de6b:/working# exit
exit

```

I have also tried to do

```
docker run \
    -e QT_XCB_GL_INTEGRATION=xcb_egl \
    -e DISPLAY=:0 \
    -v /tmp/.X11-unix:/tmp/.X11-unix \
    --gpus all \
    --privileged \
    -it colmap/colmap:latest 

```
and with this again `colmap gui`

and I get

```
root@fb14dbbe1922:/# colmap gui
No protocol specified
qt.qpa.xcb: could not connect to display :0
qt.qpa.plugin: Could not load the Qt platform plugin "xcb" in "" even though it was found.
This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.

Available platform plugins are: eglfs, linuxfb, minimal, minimalegl, offscreen, vnc, xcb.

*** Aborted at 1706784780 (unix time) try "date -d @1706784780" if you are using GNU date ***
PC: @                0x0 (unknown)
*** SIGABRT (@0x1c) received by PID 28 (TID 0x7f154febc000) from PID 28; stack trace: ***
    @     0x7f1555605046 (unknown)
    @     0x7f1553cb4520 (unknown)
    @     0x7f1553d089fc pthread_kill
    @     0x7f1553cb4476 raise
    @     0x7f1553c9a7f3 abort
    @     0x7f1554260ba3 QMessageLogger::fatal()
    @     0x7f1554867713 QGuiApplicationPrivate::createPlatformIntegration()
    @     0x7f1554867c08 QGuiApplicationPrivate::createEventDispatcher()
    @     0x7f1554490b17 QCoreApplicationPrivate::init()
    @     0x7f155486ab70 QGuiApplicationPrivate::init()
    @     0x7f1554f7eced QApplicationPrivate::init()
    @     0x5618a3eb59ad colmap::RunGraphicalUserInterface()
    @     0x5618a3e781ab main
    @     0x7f1553c9bd90 (unknown)
    @     0x7f1553c9be40 __libc_start_main
    @     0x5618a3e7e945 _start
Aborted (core dumped)

Is there something I need to do to be able to use the GUI?
```

**Comment by KansaiUser** [comment]
I managed to run the GUI with

```
xhost +local:root

docker run --rm -it \
    --volume="/tmp/.X11-unix:/tmp/.X11-unix:rw" \
    --env="QT_X11_NO_MITSHM=1" \
    --env="DISPLAY"  \
    -w /working \
    -v $1:/working  \
    --gpus all \
    --privileged \
    colmap/colmap:latest 
```

Not sure if it is correct or not

**Comment by xyh1233** [comment]
Did you solve this problem?

root@01fc4655de6b:/working# colmap gui
qt.qpa.xcb: could not connect to display 
qt.qpa.plugin: Could not load the Qt platform plugin "xcb" in "" even though it was found.
This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.

Available platform plugins are: eglfs, linuxfb, minimal, minimalegl, offscreen, vnc, xcb.

*** Aborted at 1706784366 (unix time) try "date -d @1706784366" if you are using GNU date ***
PC: @                0x0 (unknown)
*** SIGABRT (@0x1b) received by PID 27 (TID 0x7f379e015000) from PID 27; stack trace: ***
    @     0x7f37a3749046 (unknown)
    @     0x7f37a1dfc520 (unknown)
    @     0x7f37a1e509fc pthread_kill
    @     0x7f37a1dfc476 raise
    @     0x7f37a1de27f3 abort
    @     0x7f37a23a6ba3 QMessageLogger::fatal()
    @     0x7f37a29ad713 QGuiApplicationPrivate::createPlatformIntegration()
    @     0x7f37a29adc08 QGuiApplicationPrivate::createEventDispatcher()
    @     0x7f37a25d6b17 QCoreApplicationPrivate::init()
    @     0x7f37a29b0b70 QGuiApplicationPrivate::init()
    @     0x7f37a30c4ced QApplicationPrivate::init()
    @     0x560ef1d759ad colmap::RunGraphicalUserInterface()
    @     0x560ef1d381ab main
    @     0x7f37a1de3d90 (unknown)
    @     0x7f37a1de3e40 __libc_start_main
    @     0x560ef1d3e945 _start
Aborted (core dumped)
root@01fc4655de6b:/working# exit
exit
